homeassistant/components/pooldose/__init__.py

```diff
@@ -0,0 +1,72 @@
+"""The Seko Pooldose integration."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from pooldose.client import PooldoseClient
+from pooldose.request_handler import RequestStatus
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_SCAN_INTERVAL, CONF_TIMEOUT, Platform
+from homeassistant.core import HomeAssistant
+
+from .const import CONF_INCLUDE_SENSITIVE_DATA, DEFAULT_SCAN_INTERVAL, DEFAULT_TIMEOUT
+from .coordinator import PooldoseCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+_PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+"""Configure the Seko PoolDose entry."""
+
+
+async def async_update_device_info(client: PooldoseClient) -> dict[str, str | None]:
+    """Fetch latest device info from all relevant endpoints."""
+    device_info: dict[str, str | None] = {}
+    if client.device_info is None:
+        _LOGGER.error("Device info is not available from PoolDose client")
+    else:
+        device_info = client.device_info
+    return device_info
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Seko PoolDose from a config entry."""
+    # Obtain values, preferring options (re‑configure) over static data
+    host = entry.options.get(CONF_HOST, entry.data[CONF_HOST])
+    scan_interval = entry.options.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
+    timeout = entry.options.get(CONF_TIMEOUT, DEFAULT_TIMEOUT)
+    include_sensitive_data = entry.options.get(
+        CONF_INCLUDE_SENSITIVE_DATA, entry.data.get(CONF_INCLUDE_SENSITIVE_DATA, False)
+    )
+
+    # Create the PoolDose API client
+    client_status, client = await PooldoseClient.create(
+        host, timeout, include_sensitive_data
+    )
```

Comment:
I would recommend to split the creation of your client from doing a request. This makes testing a lot easier